File: FluentFTP.CSharpExamples/OpenWrite.cs
using System;
using System.Net;
using System.Threading;
using System.Threading.Tasks;
using FluentFTP;

namespace Examples {
	internal static class OpenWriteExample {

		public static void OpenWrite() {
			using (var conn = new FtpClient("127.0.0.1", "ftptest", "ftptest")) {
				conn.Connect();

				// open an write-only stream to the file
				using (var ostream = conn.OpenWrite("/full/or/relative/path/to/file")) {
					try {
						// ostream.Position is incremented accordingly to the writes you perform
					}
					finally {
						ostream.Close();
					}
				}
			}
		}

		public static async Task OpenWriteAsync() {
			var token = new CancellationToken();
			using (var conn = new FtpClient("127.0.0.1", "ftptest", "ftptest")) {
				await conn.ConnectAsync(token);

				// open an write-only stream to the file
				using (var ostream = await conn.OpenWriteAsync("/full/or/relative/path/to/file", token)) {
					try {
						// ostream.Position is incremented accordingly to the writes you perform
					}
					finally {
						ostream.Close();
					}
				}
			}
		}

	}
}

File: FluentFTP.VBExamples/OpenWrite.vb
Imports System
Imports System.Net
Imports System.Threading
Imports System.Threading.Tasks
Imports FluentFTP

Namespace Examples
	Friend Module OpenWriteExample
		Sub OpenWrite()
			Using conn = New FtpClient("127.0.0.1", "ftptest", "ftptest")
				conn.Connect()

				' open an write-only stream to the file
				Using ostream = conn.OpenWrite("/full/or/relative/path/to/file")

					Try
						' ostream.Position Is incremented accordingly to the writes you perform
					Finally
						ostream.Close()
					End Try
				End Using
			End Using
		End Sub

		Async Function OpenWriteAsync() As Task
			Dim token = New CancellationToken()

			Using conn = New FtpClient("127.0.0.1", "ftptest", "ftptest")
				Await conn.ConnectAsync(token)

				' open an write-only stream to the file
				Using ostream = Await conn.OpenWriteAsync("/full/or/relative/path/to/file", token)

					Try
						' ostream.Position Is incremented accordingly to the writes you perform
					Finally
						ostream.Close()
					End Try
				End Using
			End Using
		End Function
	End Module
End Namespace
